src/entity/Location/DistanceCalculator.java

```diff
@@ -0,0 +1,89 @@
+package entity.Location;
+
+import entity.Events.Event;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.math.BigDecimal;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Arrays;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+public class DistanceCalculator implements DistanceCalculatorInterface {
+    public DistanceCalculator() {
+        ;
+    }
+
+    @Override
+    public boolean within2KM(String[] strCoord, Event event) throws Exception {
+        return calculateDistance(strCoord, event) <= 2;
+    }
+
+    public double calculateDistance(String[] startCoord, Event event) throws Exception {
+        double startLat = Double.valueOf(Arrays.stream(startCoord).toList().get(0));
+        double startLon = Double.valueOf(Arrays.stream(startCoord).toList().get(1));
+        String[] endCoord = event.getLocation().getCoordinates();
+        double endLat = Double.valueOf(Arrays.stream(endCoord).toList().get(0));
+        double endLon = Double.valueOf(Arrays.stream(endCoord).toList().get(1));
+
+        String allApiValues = apiCaller(startLat, startLon, endLat, endLon);
+        return extractDistance(allApiValues);
+    }
+
+
+    private String apiCaller(double startLat, double startLon, double endLat, double endLon) throws Exception{
+        try {
+                String apiKey = "An_Fn1bhTQPuLmUi8MsH-5_btdPIKgINhoecH-ayEq0rjUhrnFbXoHHVnwjAli_K";
+                String bingMapsApiUrl = "https://dev.virtualearth.net/REST/v1/Routes/DistanceMatrix" +
+                        "?origins=" + startLat + "," + startLon +
+                        "&destinations=" + endLat + "," + endLon +
+                        "&travelMode=driving&startTime=2023-10-30T08:00:00&timeUnit=minute&key=" + apiKey;
+                // Example URL: dev.virtualearth.net/REST/v1/Routes/DistanceMatrix?origins=37.7749,-122.4194;34.0522,-118.2437;40.7128,-74.0060&destinations=40.7128,-74.0060;41.8781,-87.6298;34.0522,-118.2437&travelMode=driving&startTime=2023-10-30T08:00:00&timeUnit=minute&key=YourBingMapsKey
+
+                URL url = new URL(bingMapsApiUrl);
+                HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+                connection.setRequestMethod("GET");
+                int responseCode = connection.getResponseCode();
+                if (responseCode == 200) {
+                    BufferedReader in = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+                    String inputLine;
+                    StringBuilder response = new StringBuilder();
+
+                    while ((inputLine = in.readLine()) != null) {
+                        response.append(inputLine);
+                    }
+                    in.close();
+
+                    return response.toString();
+                }
+            } catch (IOException e) {
+                throw new RuntimeException("API call failed");
```

Comment:
I suggest doing something else in the catch, otherwise, the entire program will stop if we throw an exception. For instance, in my API calls, I returned a null object when a map cannot be generated and it is up to the interface adapters to appropriately deal with a null object, such as informing the user about the error.